janitor/functions.py

```diff
@@ -2408,3 +2408,21 @@ def impute(df, column: str, value=None, statistic=None):
     if value is not None:
         df[column] = df[column].fillna(value)
     return df
+
+
+@pf.register_dataframe_method
+def then(df: pd.DataFrame, func) -> pd.DataFrame:
```

Comment:
Wondering if we should allow for `*args, **kwargs` in here?

Reply:
+1

Reply:
I don't think so, since the purpose is to pass a one-argument function to `then`. It's similar to `apply` where any parameters or such should be done in the referenced function.

Unless you guys think otherwise.

Reply:
I would personally like it, but then I think to prevent premature overengineering, I'd just merge this version first, and we can add in args/kwargs later.

Reply:
Could always do:
```python
def func(df, thing1, thing2):
    dostuff()

df.blah().then(lambda df: func(df, 1, 'asdf'))
```

if you want to use extra args for now.

Reply:
So I'm a bit torn...

I like that it doesn't have arguments so it's forced to read like English: `.then(do_this_to_the_df)`, but arguments obviously makes it more flexible.

However, one can always use `functools.partial` to coerce a function to a one-argument function (as I usually do with `pd.Series.apply`).

So the trade off would be: arguments in `.then()` or force people to use `functools.partial`?

Reply:
`functools.partial` is pretty intermediate-level stuff!